### PR TITLE
[JSDoc] Add missing documentation for botbuilder-dialogs-adaptive root

### PR DESCRIPTION
**Pull request by sw-joelmut**
Addresses # 2602

## Description
This PR adds all the missing JSDoc documentation for [botbuilder-dialogs-adaptive](https://github.com/microsoft/botbuilder-js/tree/master/libraries/botbuilder-dialogs-adaptive) based on the errors thrown. Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet)

We divided the work for this library into multiple PRs to ease the review.

## Specific Changes

- Ported documentation from botbuilder-dotnet
- Added `@private` for private methods.
- Added `@protected` for protected methods.
- Added JSDoc comments in **13 files**.

## Testing
In the following image there is a sneak peek about the ESLint Report with no JSDoc errors.
image

### CODE REVIEW

**Review by iscaltritti-sw — APPROVED** [review]

File: libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts [hunk 1]
```diff
@@ -582,12 +680,23 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> {
...
+     */
     protected onSetScopedServices(dialogContext: DialogContext): void {
         if (this.generator) {
             dialogContext.services.set(languageGeneratorKey, this.generator);
         }
     }
 
+    /**
+     * @protected
+     * Removes the current most action from the given `ActionContext` if there are any.
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts:696]:
```suggestion
     * Removes the most current action from the given `ActionContext` if there are any.
```

File: libraries/botbuilder-dialogs-adaptive/src/adaptiveTypeBuilder.ts [hunk 2]
```diff
@@ -8,17 +8,30 @@
...
+ */
 export class AdaptiveTypeBuilder implements TypeBuilder {
     private _factory: new () => object;
     private _converters: { [key: string]: Converter } = {};
     private _resourceExplorer: ResourceExplorer;
 
+    /**
+     * Creates a new instance of the `AdaptiveTypeBuilder` class.
+     * @param factory Factory for the adaptive type.
+     * @param resourceExplorer  Resource explorer.
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/adaptiveTypeBuilder.ts:22]:
```suggestion
     * @param resourceExplorer Resource explorer.
```

File: libraries/botbuilder-dialogs-adaptive/src/adaptiveTypeBuilder.ts [hunk 3]
```diff
@@ -8,17 +8,30 @@
...
+     * @param converters Key value pair with converters.
+     */
     public constructor(factory: new () => object, resourceExplorer: ResourceExplorer, converters: { [key: string]: Converter }) {
         this._factory = factory;
         this._resourceExplorer = resourceExplorer;
         this._converters = converters;
     }
 
+    /**
+     * Builds a adaptive type.
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/adaptiveTypeBuilder.ts:32]:
```suggestion
     * Builds an adaptive type.
```

File: libraries/botbuilder-dialogs-adaptive/src/entityInfo.ts [hunk 4]
```diff
@@ -83,6 +83,9 @@ export interface NormalizedEntityInfos {
     [name: string]: Partial<EntityInfo>[];
 }
 
+/**
+ * Extended information about an entity including $instance data.
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/entityInfo.ts:87]: Is this ok? $instance

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Review by santgr11 — APPROVED** [review]

File: libraries/botbuilder-dialogs-adaptive/src/entityInfo.ts (hunk 4 above)
Comment (sw-joelmut) [reply libraries/botbuilder-dialogs-adaptive/src/entityInfo.ts:87]: I just copied it from the line [10](https://github.com/southworks/botbuilder-js/pull/172/files/6d532cc55bc5bcbe691a39ad1341709f8bf5467a#diff-ce80c52b5d143b729099e5997b0ddc12L10) of this file, I checked out in dotnet and it's the same comment but nothing to reference to `$instance` as well.

**Comment by sw-joelmut** [comment]
@ceciliaavila, feedback applied, thanks. Just the $instance comment remains.

File: libraries/botbuilder-dialogs-adaptive/src/entityInfo.ts (hunk 4 above)
Comment (ceciliaavila) [reply libraries/botbuilder-dialogs-adaptive/src/entityInfo.ts:87]: Ok, let's keep it then. Thanks.

**Review by ceciliaavila — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
Promoted

**Pull request closed without merging**